src/ring_theory/polynomial/symmetric.lean

```diff
@@ -141,48 +141,23 @@ end
 lemma esymm_eq_sum_monomial (n : ℕ) : esymm σ R n =
   ∑ t in powerset_len n univ, monomial (∑ i in t, finsupp.single i 1) 1 :=
 begin
-  refine sum_congr rfl (λ x hx, _),
-  rw monic_monomial_eq,
-  rw finsupp.prod_pow,
-  rw ← prod_subset (λ y _, finset.mem_univ y : x ⊆ univ) (λ y _ hy, _),
-  { refine prod_congr rfl (λ x' hx', _),
-    convert (pow_one _).symm,
-    convert (finsupp.apply_add_hom x' : (σ →₀ ℕ) →+ ℕ).map_sum _ x,
-    classical,
-    simp [finsupp.single_apply, finset.filter_eq', apply_ite, apply_ite finset.card],
-    rw if_pos, exact hx', },
-  { convert pow_zero _,
-    convert (finsupp.apply_add_hom y : (σ →₀ ℕ) →+ ℕ).map_sum _ x,
-    classical,
-    simp [finsupp.single_apply, finset.filter_eq', apply_ite, apply_ite finset.card],
-    rw if_neg, exact hy }
+  simp_rw monomial_sum_one,
+  refl,
 end
 
 @[simp] lemma esymm_zero : esymm σ R 0 = 1 :=
 by simp only [esymm, powerset_len_zero, sum_singleton, prod_empty]
 
 lemma map_esymm (n : ℕ) (f : R →+* S) : map f (esymm σ R n) = esymm σ S n :=
-begin
-  rw [esymm, (map f).map_sum],
-  refine sum_congr rfl (λ x hx, _),
-  rw (map f).map_prod,
-  simp,
-end
+by simp_rw [esymm, map_sum, map_prod, map_X]
 
 lemma rename_esymm (n : ℕ) (e : σ ≃ τ) : rename e (esymm σ R n) = esymm τ R n :=
 begin
-  rw [esymm_eq_sum_subtype, esymm_eq_sum_subtype, (rename ⇑e).map_sum],
-  let e' : {s : finset σ // s.card = n} ≃ {s : finset τ // s.card = n} :=
-    equiv.subtype_equiv (equiv.finset_congr e) (by simp),
-  rw ← equiv.sum_comp e'.symm,
-  apply fintype.sum_congr,
-  intro,
-  calc _ = (∏ i in (e'.symm a : finset σ), (rename e) (X i)) : (rename e).map_prod _ _
-     ... = (∏ i in (a : finset τ), (rename e) (X (e.symm i))) : prod_map (a : finset τ) _ _
-     ... = _ : _,
-  apply finset.prod_congr rfl,
-  intros,
-  simp,
+  have : univ.map e.to_embedding = univ, { ext, simp },
+  simp_rw [esymm, map_sum, map_prod, rename_X, ←equiv.to_embedding_apply, ←finset.prod_map,
+    ←equiv.finset_congr_apply, ←equiv.to_embedding_apply,
+    ←finset.sum_map _ _ (λ x, finset.prod x (X : τ → mv_polynomial τ R)),
+    equiv.finset_congr_to_embedding, ←finset.powerset_len_map, this],
```

Comment:
Could change this to
```lean
calc rename e (esymm σ R n)
     = ∑ x in powerset_len n univ, ∏ i in x, X (e i)
       : by simp_rw [esymm, map_sum, map_prod, rename_X]
 ... = ∑ t in powerset_len n (univ.map e.to_embedding), ∏ i in t, X i
       : by simp [finset.powerset_len_map]
 ... = ∑ t in powerset_len n univ, ∏ i in t, X i :
    have univ.map e.to_embedding = univ, { ext, simp }, by simp_rw this
```
if it's preferred

Reply:
I think that calc proof is easier to decipher, so it has my vote!

Reply:
Otherwise LGTM. Thanks!
bors d+